addon/-record-data-private/system/model/internal-model.js

```diff
@@ -315,24 +318,35 @@ export default class InternalModel {
     this._doNotDestroy = false;
 
     if (this._record) {
-      Object.keys(this._relationshipPromisesCache).forEach(key => {
-        // TODO Igor cleanup the guard
-        // TODO there is probably relationship cleanup to do outside of the _record check
-        if (this._relationshipPromisesCache[key].destroy) {
-          this._relationshipPromisesCache[key].destroy();
-        }
-        delete this._relationshipPromisesCache[key];
-      });
-      Object.keys(this._manyArrayCache).forEach(key => {
-        this._retainedManyArrayCache[key] = this._manyArrayCache[key];
-        delete this._manyArrayCache[key];
-      });
       this._record.destroy();
     }
 
+    Object.keys(this._relationshipPromisesCache).forEach(key => {
+      // TODO Igor cleanup the guard
+      if (this._relationshipPromisesCache[key].destroy) {
+        this._relationshipPromisesCache[key].destroy();
+      }
+      delete this._relationshipPromisesCache[key];
+    });
+    // if the manyArray is for a sync relationship
+    //   we should clear it
+    Object.keys(this._manyArrayCache).forEach(key => {
+      let manyArray = (this._retainedManyArrayCache[key] = this._manyArrayCache[key]);
+      delete this._manyArrayCache[key];
+
+      /*
+          It is likely in this case instead of retaining we should destroy
+
+          - @runspired
+        */
+      if (manyArray && !manyArray._inverseIsAsync) {
+        manyArray.clear();
```

Comment:
What's the motivation for this?

Reply:
Sync relationships act as client-side delete. Previously, in this case the manyArray would be cleared by the relationship in this scenario, but with the refactoring for `RecordData` the manyArray is now managed by `InternalModel` and thus `InternalModel` must clear it in this case.